test/library/driven_freq_tuning/test_stark_p1_spect.py

```diff
@@ -186,71 +207,98 @@ def test_running_analysis_without_service(self):
         np.testing.assert_array_almost_equal(test_xvals, ref_xvals)
 
-    def test_running_analysis_with_service(self):
+    @named_data(
+        ["ordinary", 5e6, 200e6, -50e6, 5e6, -180e6, -40e6, 100e3],
+        ["asymmetric_inflection_1st_ord", 10e6, 200e6, -20e6, -50e6, -180e6, -20e6, -10e6],
+        ["inflection_3st_ord", 10e6, 200e6, -80e6, 80e6, -180e6, -200e6, 100e3],
+    )
+    @unpack
+    def test_running_analysis_with_service(self, po1, po2, po3, no1, no2, no3, ferr):
         """Test running analysis by setting service to the experiment data.
 
         This must convert x-axis into frequencies with the Stark coefficients.
         """
+        mock_experiment_id = "6453f3d1-04ef-4e3b-82c6-1a92e3e066eb"
+        mock_result_id = "d067ae34-96db-4e8e-adc8-030305d3d404"
+        mock_backend = FakeHanoiV2().name
+
+        coeffs = util.StarkCoefficients(
+            pos_coef_o1=po1,
+            pos_coef_o2=po2,
+            pos_coef_o3=po3,
+            neg_coef_o1=no1,
+            neg_coef_o2=no2,
+            neg_coef_o3=no3,
+            offset=ferr,
+        )
+
+        service = FakeService()
+        service.create_experiment(
+            experiment_type="StarkRamseyXYAmpScan",
+            backend_name=mock_backend,
+            experiment_id=mock_experiment_id,
+        )
+        service.create_analysis_result(
+            experiment_id=mock_experiment_id,
+            result_data={"value": coeffs},
+            result_type="stark_coefficients",
+            device_components=["Q0"],
+            tags=[],
+            quality="Good",
+            verified=False,
+            result_id=mock_result_id,
+        )
+
         analysis = StarkP1SpectAnalysisReturnXvals()
 
         xvals = np.linspace(-1, 1, 11)
+        ref_fvals = util.convert_amp_to_freq(xvals, coeffs)
+
         exp_data = ExperimentData(
-            service=self.service,
+            service=service,
             backend=FakeHanoiV2(),
         )
         exp_data.metadata.update({"physical_qubits": [0]})
         for x in xvals:
             exp_data.add_data({"counts": {"0": 1000, "1": 0}, "metadata": {"xval": x}})
-        analysis.run(exp_data, replace_results=True)
-        test_xvals = exp_data.analysis_results("xvals").value
-        ref_xvals = np.where(
-            xvals > 0,
-            (
-                self.coeffs["stark_pos_coef_o1"] * xvals
-                + self.coeffs["stark_pos_coef_o2"] * xvals**2
-                + self.coeffs["stark_pos_coef_o3"] * xvals**3
-                + self.coeffs["stark_ferr"]
-            ),
-            (
-                self.coeffs["stark_neg_coef_o1"] * xvals
-                + self.coeffs["stark_neg_coef_o2"] * xvals**2
-                + self.coeffs["stark_neg_coef_o3"] * xvals**3
-                + self.coeffs["stark_ferr"]
-            ),
-        )
-        np.testing.assert_array_almost_equal(test_xvals, ref_xvals)
+        analysis.run(exp_data, replace_results=True).block_for_results()
```

Comment:
It doesn't matter, but `ExperimentData.analysis_results()` blocks by default so you don't need to block here. At one point, I started going through the tests adding `block_for_results` before realizing this because I was worried about race conditions.